s4/clarity/artifact.py

```diff
@@ -213,5 +214,100 @@ def reagent_label_name(self, reagent_label_name):
         reagent_label = self.make_subelement_with_parents("./reagent-label")
         reagent_label.set("name", reagent_label_name)
 
+    @lazy_property
+    def demux(self):
+        # type: () -> ArtifactDemux
+        """
+        Provides access to the 'demux' endpoint added in Clarity 5.1.
+
+        Using this property with an earlier version of Clarity will result in
+        a 404 Not Found error.
+
+        :type: ArtifactDemux
+        """
+        return ArtifactDemux(self.lims, self.uri + "/demux")
+
     def _get_attach_to_key(self):
         return self.type, ""
+
+
+class DemuxArtifact(WrappedXml):
+    """
+    Corresponds to the 'demux-artifact' type in the 'artifact' namespace of the Clarity API.
+    """
+    artifact = subnode_link(Artifact, ".")
+    reagent_labels = subnode_element_list(ReagentLabel, "reagent-labels", "reagent-label", readonly=True)
+
+    @property
+    def demux(self):
+        # type: () -> DemuxDetails
+        """
+        The element `DemuxDetails` from subnode 'demux'
+
+        Some versions of Clarity do not provide this element when the `DemuxArtifact`
+        is a pool of artifacts that are all from the same submitted sample. In this
+        case, if more than one artifact is labeled, the pool `Artifact`'s demux will
+        be provided, but if only one artifact is labeled, the pool's demux will not
+        be provided.
```

Comment:
Added this to address the edge case I noticed while writing the comment below.

I'm not sure if it's clear enough that a `DemuxArtifact` that is a pool is a sub-pool in a pool of pools.